### PR TITLE
Update lightning_cli_advanced_2.rst

### PR DESCRIPTION
**Pull request by fabiofumarola**
In order to allow variable interpolation we need to add kwargs to the parser. I've checked the code and this is the only way to let the CLI to call Omegaconf.resolve

## What does this PR do?

Fixes #\<issue_number>

### Does your PR introduce any breaking changes? If yes, please list them.

No

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @mauvilsa @carmocca

### CODE REVIEW

**Comment by mauvilsa** [comment]
Minor comment. This fix is already in pull request #14976, see https://github.com/Lightning-AI/lightning/pull/14976/files#diff-4854de42e45c247907e68157a6460964fd365449de9e3f599982cc05b330260fR186-R192. I don't have anything against merging this pull request and remove the change in the other one. But do note that this is only one of many mistakes that the documentation currently has.

@fabiofumarola I would suggest to remove `seed_everything_default=True`. The default value of that parameter is anyway `True` and it distracts from what that part of the documentation is explaining.

File: docs/source-pytorch/cli/lightning_cli_advanced_2.rst [hunk 1]
```diff
@@ -200,7 +200,15 @@ To enable variable interpolation, first install omegaconf:
...
+
+.. code: python
+    cli = LightningCLI(
+        seed_everything_default=True,
+        parser_kwargs={"parser_mode": "omegaconf"},
+    )
+
+Specify the `parser_kwargs as above` then Lightning CLI will automatically handle variables in yaml files:
 
 .. code bash:
```
Comment (carmocca) [inline docs/source-pytorch/cli/lightning_cli_advanced_2.rst:213]:
```suggestion
Then set omegaconf when instantiating the ``LightningCLI`` class:

.. code:: python
    cli = LightningCLI(MyModel, parser_kwargs={"parser_mode": "omegaconf"})
After this, the CLI will automatically perform interpolation in yaml files:

.. code:: bash
```

**Review by carmocca — APPROVED** [review]
Added Mauricio as co-author in the merge message

**Review by awaelchli — APPROVED** [review]

File: docs/source-pytorch/cli/lightning_cli_advanced_2.rst [hunk 2]
```diff
@@ -200,8 +200,13 @@ To enable variable interpolation, first install omegaconf:
 
     pip install omegaconf
 
-Once this is installed, the Lightning CLI will automatically handle variables in yaml files:
+Then set omegaconf when instantiating the ``LightningCLI`` class:
 
-.. code bash:
+.. code:: python
+    cli = LightningCLI(MyModel, parser_kwargs={"parser_mode": "omegaconf"})
```
Comment (carmocca) [inline docs/source-pytorch/cli/lightning_cli_advanced_2.rst:206]:
```suggestion

    cli = LightningCLI(MyModel, parser_kwargs={"parser_mode": "omegaconf"})
```

**Pull request merged**